asyncio_mqtt/client.py

```diff
@@ -539,13 +579,11 @@ def _on_subscribe(
         client: mqtt.Client,
         userdata: Any,
         mid: int,
-        granted_qos: int,
+        granted_qos: Union[Tuple[int], List[mqtt.ReasonCodes]],
         properties: Optional[mqtt.Properties] = None,
     ) -> None:
         try:
-            fut = self._pending_subscribes.pop(mid)
-            if not fut.done():
-                fut.set_result(granted_qos)
+            self._pending_subscribes.pop(mid).set()
```

Comment:
See above. Revert for now. It's not time for a breaking change. Also, the `if` may seem redundant but it actually avoids a race condition. Trust me. :)